sklearn/decomposition/_pca.py

```diff
@@ -177,26 +180,42 @@ class PCA(_BasePCA):
         improve the predictive accuracy of the downstream estimators by
         making their data respect some hard-wired assumptions.
 
-    svd_solver : {'auto', 'full', 'arpack', 'randomized'}, default='auto'
-        If auto :
+    svd_solver : {'auto', 'full', 'covariance_eigh', 'arpack', 'randomized'},\
+            default='auto'
+        "auto" :
             The solver is selected by a default policy based on `X.shape` and
-            `n_components`: if the input data is larger than 500x500 and the
-            number of components to extract is lower than 80% of the smallest
-            dimension of the data, then the more efficient 'randomized'
-            method is enabled. Otherwise the exact full SVD is computed and
-            optionally truncated afterwards.
-        If full :
-            run exact full SVD calling the standard LAPACK solver via
+            `n_components`: if the input data has fewer than 1000 features and
+            more than 10 times as many samples, then the "covariance_eigh"
+            solver is used. Otherwise, if the input data is larger than 500x500
+            and the number of components to extract is lower than 80% of the
+            smallest dimension of the data, then the more efficient
+            "randomized" method is enabled. Otherwise the exact "full" SVD is
+            computed and optionally truncated afterwards.
+        "full" :
+            Run exact full SVD calling the standard LAPACK solver via
             `scipy.linalg.svd` and select the components by postprocessing
-        If arpack :
-            run SVD truncated to n_components calling ARPACK solver via
+        "covariance_eigh" :
+            Precompute the covariance matrix (on centered data), run a
+            classical eigenvalue decomposition on the covariance matrix
+            typically using LAPACK and select the components by postprocessing.
+            This solver is very efficient when the number of features is small
+            and not tractable otherwise (large memory footprint required to
+            materialize the covariance matrix). Also note that compared to the
```

Comment:
```suggestion
            This solver is very efficient for n_samples >> n_features and small n_features.
            It is, however, not tractable otherwise for large n_features (large memory footprint required to
            materialize the covariance matrix). Also note that compared to the
```
Do we know the computation or memory complexity in terms of n_features, like memory is `O(n_features**2)` and computation `O(n_features**3)`?

Reply:
I know that it's very bad empirically (see the benchmarks above). The memory is definitely `O(n_features**2)`. For computation I am not 100% sure but indeed it should be dominated by the `eigh` call on a `n_features` square matrix, so it should indeed be cubic w.r.t. `n_features`.